src/systems/enemy.rs

```diff
@@ -52,18 +64,18 @@ impl<'s> System<'s> for EnemySystem {
             //transform the spaceship in x and y by the currrent velocity in x and y
             enemy_component.update_position(enemy_transform, time.delta_seconds(), enemy_motion);
 
-            enemy_component.health -= enemy_component.poison;
+            enemy_health.health -= enemy_component.poison;
+            enemy_health.constrain();
 
             //conditions for despawning
             if enemy_transform.translation()[1] + enemy_hitbox.height / 2.0 < ARENA_MIN_Y {
-                //defense is damage is enemy gets past
-                for defense in (&mut defenses).join() {
-                    defense.defense -= enemy_component.defense_damage;
-                }
+                //defense is damaged if enemy gets past
```

Comment:
super small nit: add whitespace between "//" and "defense" 
```suggestion
                // defense is damaged if enemy gets past
```